### DISCUSSION TITLE
HelmChart reconciliate strategy

### DISCUSSION THREAD

**Discussion by fpiras89** [fluxcd/flux2#4622] [Q&A]
Hi, is there a way to limit HelmChart reconciliate to specific source paths from a GitRepository in particular when reconcileStrategy is set to Revision? Or is it something you planned to implement? 

I have a problem similar to the one described here [2918](https://github.com/fluxcd/flux2/discussions/2918). 

I have one Git repository that hosts multiple charts. I created multiple GitRepository resources (one per chart) with ignore configuration to limit the chart artifact to specific paths but everytime I make a change to one charts, also the other charts are reconciliated. 

Thanks!

**Comment by souleb** [comment]
I suggest you open an issue for this, with an example so we can reproduce it.

**Comment by stefanprodan** [comment]
> I have one Git repository that hosts multiple charts. I created multiple GitRepository resources (one per chart) with ignore configuration to limit the chart artifact to specific paths but everytime I make a change to one charts, also the other charts are reconciliated.

Due to the nature of Git, there is no way we can prevent the reconciliation, since there is a single revision (the commit SHA) for all charts. Using Git for charts is meant to be used for development and testing of your pipelines. For production system you should publish your charts to a container registry and use Flux OCI HelmRepos.

**Comment by fpiras89** [comment]
Hi, thanks for the quick reply!

I understand that the Git nature doesn't allow you out-of-the-box to partially fetch parts of a repository. 

By the way, from my understanding, you are creating HelmChart artifacts using only parts of a repository according to the .spec.ignore configuration of a GitRepository resource used as a source for a HelmChart. 

Would it be possible to use the same approach to trigger a reconciliation only if the parts specified in .spec.ignore changed? Maybe, to not introduce a breaking change, it would be good to introduce a new reconcileStrategy like "RevisionAndPaths" that uses the .spec.ignore to understand which files should be monitored.
